grpc/src/main/java/io/stargate/grpc/service/QueryHandler.java

```diff
@@ -102,8 +113,19 @@ protected CompletionStage<ResponseAndTraceId> buildResponse(Result result) {
                 });
       case Rows:
         try {
+          Result.Rows rows = (Result.Rows) result;
+          RowDecorator rowDecorator = null;
+          if (this.enrichResponse) {
+            rowDecorator = connection.makeRowDecorator(TableName.of(rows.resultMetadata.columns));
+          }
           responseBuilder.setResultSet(
-              ValuesHelper.processResult((Result.Rows) result, message.getParameters()));
+              ValuesHelper.processResult(
+                  rows,
+                  message.getParameters(),
+                  this::getComparableBytesFromRow,
+                  this::getPagingStateFromRow,
+                  this::makeRow,
+                  rowDecorator));
```

Comment:
nit: should we have an if else for the enriched stuff, seems cleaner and more readble:

```
if (enrichResponse) {
    RowDecorator rowDecorator = connection.makeRowDecorator(TableName.of(rows.resultMetadata.columns));
    responseBuilder.setResultSet(ValuesHelper.processResult(
          rows,
          message.getParameters(),
          this::getComparableBytesFromRow,
          this::getPagingStateFromRow,
          this::makeRow,
          rowDecorator)));
} else {
      responseBuilder.setResultSetValuesHelper.processResult(rows, message.getParameters()));
}
```